connectors/webhook/src/main/java/io/camunda/connector/inbound/HttpWebhookExecutable.java

```diff
@@ -83,65 +86,99 @@
     }
+  }
+
+  private static MappedHttpRequest mapRequest(WebhookProcessingPayload payload) {
+    return new MappedHttpRequest(
+        HttpWebhookUtil.transformRawBodyToMap(
+            payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers())),
+        payload.headers(),
+        payload.params());
+  }
 
-    WebhookProcessingResultImpl response = new WebhookProcessingResultImpl();
+  @Nullable
+  private Function<WebhookResultContext, WebhookHttpResponse> mapResponseExpression() {
+    Function<WebhookResultContext, WebhookHttpResponse> responseExpression = null;
+    if (props.responseExpression() != null) {
+      responseExpression = props.responseExpression();
+    } else if (props.responseBodyExpression() != null) {
+      // To be backwards compatible we need to wrap the responseBodyExpression into a
+      // responseExpression
+      // and only use the body in the final response
+      responseExpression =
+          (context) -> {
+            Object responseBody = props.responseBodyExpression().apply(context);
+            return new WebhookHttpResponse(responseBody, null, 200);
+          };
+    }
+    return responseExpression;
+  }
 
+  private void verifySignature(WebhookProcessingPayload payload) {
     if (!webhookSignatureIsValid(payload)) {
       throw new WebhookSecurityException(
           HttpResponseStatus.UNAUTHORIZED.code(),
           Reason.INVALID_SIGNATURE,
           "HMAC signature check didn't pass");
     }
-
-    var authResult = authChecker.checkAuthorization(payload);
-    if (authResult instanceof Failure failureResult) {
-      throw failureResult.toException();
-    }
-
-    response.setRequest(
-        new MappedHttpRequest(
-            HttpWebhookUtil.transformRawBodyToMap(
-                payload.rawBody(), HttpWebhookUtil.extractContentType(payload.headers())),
-            payload.headers(),
-            payload.params()));
-
-    if (props.responseBodyExpression() != null) {
-      response.setResponseBodyExpression(props.responseBodyExpression());
-    }
-
-    return response;
   }
 
-  private boolean webhookSignatureIsValid(WebhookProcessingPayload payload)
-      throws NoSuchAlgorithmException, InvalidKeyException, IOException {
-    if (shouldValidateHmac()) {
-      HMACEncodingStrategy strategy =
-          HMACEncodingStrategyFactory.getStrategy(props.hmacScopes(), payload.method());
-      byte[] bytesToSign = strategy.getBytesToSign(payload);
-      return validateHmacSignature(bytesToSign, payload);
+  private boolean webhookSignatureIsValid(WebhookProcessingPayload payload) {
+    try {
+      if (shouldValidateHmac()) {
+        HMACEncodingStrategy strategy =
+            HMACEncodingStrategyFactory.getStrategy(props.hmacScopes(), payload.method());
+        byte[] bytesToSign = strategy.getBytesToSign(payload);
+        return validateHmacSignature(bytesToSign, payload);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
I don't have much context, but if some error is thrown here, does that make sense to consider the signature invalid and `return false;` instead of propagating the exception? Or is it a real error case that could be solved otherwise?

Reply:
I just moved the code and did not want to introduce additional changes in this PR.